src/axom/core/Array.hpp

```diff
@@ -426,6 +441,35 @@ class Array : public ArrayBase<T, DIM, Array<T, DIM, SPACE>>
   template <typename... Args>
   ArrayIterator emplace(ArrayIterator pos, Args&&... args);
 
+  /*!
+   * \brief Push a value to the back of the array.
+   *
+   * \param [in] value the value to be added to the back.
+   *
+   * \note Reallocation is done if the new size will exceed the capacity.
```

Comment:
Please add a note about the `DIM==1` restriction to the doxygen for these three methods, 
(perhaps using `\pre` (?))